protocols/request-response/tests/ping.rs

```diff
@@ -38,6 +38,38 @@ use rand::{self, Rng};
 use std::{io, iter};
 use std::{collections::HashSet, num::NonZeroU16};
 
+/// Check if is_response_outbound works properly
+#[test]
+fn pending_connections() {
+    let ping = Ping("ping".to_string().into_bytes());
+    let offline_peer = PeerId::random();
+
+    let protocols = iter::once((PingProtocol(), ProtocolSupport::Full));
+    let cfg = RequestResponseConfig::default();
+
+    let (peer1_id, trans) = mk_transport();
+    let ping_proto1 = RequestResponse::throttled(PingCodec(), protocols.clone(), cfg.clone());
+    let mut swarm1 = Swarm::new(trans, ping_proto1, peer1_id.clone());
+
+
+    let peer1 = async {
+        let request_id1 = swarm1.send_request(&offline_peer, ping.clone());
+
+        match swarm1.next().await {
+            RequestResponseEvent::Message{
```

Comment:
Hi @mxinden ,
I'm trying to write this `match` arm, but I am unable to figure out how to get around the following error:
```bash
    Checking libp2p-request-response v0.9.1 (/home/user1/rust-libp2p/protocols/request-response)
error[E0308]: mismatched types
  --> protocols/request-response/tests/ping.rs:59:13
   |
58 |           match swarm1.next().await {
   |                 ------------------- this expression has type `libp2p_request_response::throttled::Event<Ping, Pong, libp2p_request_response::throttled::codec::Message<Pong>>`
59 | /             RequestResponseEvent::Message{
60 | |                 peer,
61 | |                 message: RequestResponseMessage::Response { request_id, response }
62 | |             } => {
   | |_____________^ expected enum `libp2p_request_response::throttled::Event`, found enum `libp2p_request_response::RequestResponseEvent`
   |
   = note: expected enum `libp2p_request_response::throttled::Event<Ping, Pong, libp2p_request_response::throttled::codec::Message<Pong>>`
              found enum `libp2p_request_response::RequestResponseEvent<_, _, _>`
```